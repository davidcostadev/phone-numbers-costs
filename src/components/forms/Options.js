import React from 'react';
import PropTypes from 'prop-types';

const Options = ({ options }) => {
  if (!options.length) {
    return null;
  }

  return options.map(option => (
    <option key={option.value} value={option.value}>{option.text}</option>
  ));
};

Options.propTypes = {
  options: PropTypes.array,
};

Options.defaultProps = {
  options: [],
};

export default Options;
