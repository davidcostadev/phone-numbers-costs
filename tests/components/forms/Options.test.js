import React from 'react';
import renderer from 'react-test-renderer';
import Options from '../../../src/components/forms/Options';

describe('Options Component', () => {
  let options;

  beforeEach(() => {
    options = [
      { value: 1, text: 'One' },
      { value: 2, text: 'Two' },
    ];
  });

  describe('Render', () => {
    it('default', () => {
      const tree = renderer.create((
        <Options />
      )).toJSON();

      expect(tree).toMatchSnapshot();
    });

    it('with props', () => {
      const tree = renderer.create((
        <Options options={options} />
      )).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
});
