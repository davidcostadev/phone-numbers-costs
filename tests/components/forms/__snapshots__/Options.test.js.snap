// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Options Component Render default 1`] = `null`;

exports[`Options Component Render with props 1`] = `
Array [
  <option
    value={1}
  >
    One
  </option>,
  <option
    value={2}
  >
    Two
  </option>,
]
`;
